lib/actions/build.coffee

```diff
@@ -0,0 +1,81 @@
+# Imported here because it's needed for the setup
+# of this action
+dockerUtils = require('../utils/docker')
+
+exports.getBundleInfo = getBundleInfo = (options) ->
+	Promise = require('bluebird')
+	helpers = require('../utils/helpers')
+
+	if options.application?
+		# An application was provided
+		Promise.try(-> helpers.getAppInfo(options.application))
+		.then (app) ->
+			return [app.arch, app.device_type]
+	else if options.arch? and options.deviceType?
+		return Promise.resolve([options.arch, options.deviceType])
+	else
+		# No information, cannot do resolution
+		return
```

Comment:
Right, or `return Promise.resolve()` can be used, but the `method` is neater, can save two `resolve's`. Also I would throw here as we cannot handle this case anyway, and rely on `Promise.catch` in the build runner to capture it, log the error and do the rest of the failure handling.
Right now the build runner is mixing promises with synch paradigm which is not great.